internal/registry/obj_selector.go

```diff
@@ -84,11 +92,19 @@ func (o *ObjSelector) NextObject() (*ObjectInfo, error) {
 }
 
 // Resets object selector to start scanning objects from the beginning.
-func (o *ObjSelector) Reset() {
+// After resetting the selector is locked for specified lockTime to prevent
+// concurrency issues.
+func (o *ObjSelector) Reset(lockTime int) bool {
 	o.mu.Lock()
 	defer o.mu.Unlock()
 
+	if time.Now().UTC().Before(o.lockedUntil) {
+		return false
+	}
+
 	o.lastId = 0
+	o.lockedUntil = time.Now().UTC().Add(time.Duration(lockTime) * time.Second)
```

Comment:
Not sure you need `UTC()`. These locks have nothing to do with time zones, just duration.